code/functions.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import holoviews as hv
+from holoviews import plotting
+from holoviews import Options, Store
+import pfaffian as pf
+import kwant
+from types import SimpleNamespace
+
+
+kron_prods = ['s0s0', 's0sx', 's0sy', 's0sz',
+              'sxs0', 'sxsx', 'sxsy', 'sxsz',
+              'sys0', 'sysx', 'sysy', 'sysz',
+              'szs0', 'szsx', 'szsy', 'szsz']
+
+__all__ = ['sigma0', 'sigmax', 'sigmay', 'sigmaz', 'spectrum',
+           'plot_bands', 'h_k', 'plot_bands', 'find_pfaffian',
+           'kitaev_chain', 'spinful_kitaev_chain', 'nanowire_chain'] + kron_prods
+
+options = Store.options(backend='matplotlib')
```

Comment:
`options` are never used in follow-up code. Is this a global object that changes holoviews behavior? If yes, then it must go into `init_mooc_nb.init_notebook` function. Otherwise this violates the rule that importing a module shouldn't have any side effects (`matplotlib.pyplot` is a notorious example of bad behavior).